json/src/de.rs

```diff
@@ -523,6 +543,10 @@ impl<R: Read> DeserializerImpl<R> {
     }
 }
 
+fn stack_overflow() -> ErrorCode {
+    ErrorCode::Custom("recursion limit exceeded".into())
```

Comment:
It's a shame we can't add error codes in a backwards compatible fashion. Next major release though :)